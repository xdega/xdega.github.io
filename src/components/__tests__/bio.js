import React from "react"
import { render } from "@testing-library/react"
import Bio from "../bio"

describe("<Bio /> component", () => {
  const originalWarn = console.warn.bind(console.warn);

  beforeAll(() => {
    console.warn = (msg) => {
      /**
       * Disabling the image loading warning. Unable to find good documentation 
       * on how to propery mock a component that contains images, but these 
       * tests do not care about images per se.
       * */
      !msg.toString().includes('Image not loaded') && originalWarn(msg);
    }
  });

  afterAll(() => {
    console.warn = originalWarn
  });

  it("should render intro text correctly", () => {
    const { getByTestId } = render(<Bio />);
  
    expect(getByTestId("bio-intro"))
      .toHaveTextContent("I am a Software Engineer focused on Ed Tech and the WWW.")
  })
});