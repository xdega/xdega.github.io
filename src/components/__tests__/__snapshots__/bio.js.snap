// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bio renders correctly 1`] = `
<aside
  className="flex p-2 m-2 ml-0 pl-0 pt-0 mt-0 flex-col sm:flex-row"
>
  <div
    className="flex-5 text-gray-500 sm:pl-4"
  >
    <p
      className="mb-2 sm:mb-1"
    >
      Hello, I am a Software Engineer focused on Ed Tech and the WWW.
    </p>
    <p>
      My primary expertise includes modern PHP and JavaScript, along with numerous tools and technologies that saturate the world of Web Development. I hope you find my nerdy musings insightful.
    </p>
  </div>
</aside>
`;
